### PR TITLE
fix: Allow different types of EOL in note model template regex

### PR DESCRIPTION
**Pull request by dominik-oktav**
As reported in #51, the current regex doesn't work on Windows. I adapted the regex to allow `\r\n`, `\n` and `\r`.

### CODE REVIEW

File: brain_brew/representation/yaml/note_model_template.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 HTML_FILE = AnkiField("html_file")
 BROWSER_HTML_FILE = AnkiField("browser_html_file", default_value=None)
 
-html_separator_regex = r'[\n]{1,}[-]{1,}[\n]{1,}'
+html_separator_regex = r'[(\r\n|\r|\n)]{1,}[-]{1,}[(\r\n|\r|\n)]{1,}'
```
Comment (ohare93) [inline brain_brew/representation/yaml/note_model_template.py:26]:
```suggestion
html_separator_regex = r'(\r\n|\r|\n){1,}[-]{1,}(\r\n|\r|\n){1,}'
```

Shouldn't we remove the square brackets here, replacing the with braces for a capture group? Playing around with this on regex101.com it seems that `[(\r\n|\r|\n)]` is simple matching any individual character (as square brackets do in regex) including opening and closing braces 😅 

image

Removing the square brackets seems to achieve the desired effect 😄

**Review by ohare93 — COMMENTED** [review]
Thanks for the efforts mate. Just one comment 👍

File: brain_brew/representation/yaml/note_model_template.py (hunk 1 above)
Comment (dominik-oktav) [reply brain_brew/representation/yaml/note_model_template.py]: Oh, seems like I overlooked that. You are right of course. However, the groups need to be declared non-capturing to not get more than 1 match. I added these changes accordingly.

Comment (aplaice) [reply brain_brew/representation/yaml/note_model_template.py]: I haven't tested and I haven't used regexes for a while, but I believe that capturing vs. non-capturing only affects whether the contents of the group can be accessed later.

**Comment by helitopia** [comment]
Hi everyone,

Any updates on the PR?

Also, is there a reason the regex `(?:\r?\n){1,}[-]{1,}(?:\r?\n){1,}` is not used? (It is either `\n` for Linux or `\r\n` for Windows)

**Comment by aplaice** [comment]
> Also, is there a reason the regex (?:\r?\n){1,}[-]{1,}(?:\r?\n){1,} is not used? (It is either \n for Linux or \r\n for Windows)

I suspect that the slightly more general regex was chosen because technically, there are/were systems where just `\r` is/was used. (Classic MacOS if I remember correctly.) This is probably more relevant in situations where one has to handle old text files, though; here, I expect we will never encounter such files since Anki itself is younger than the unix-based MacOS. OTOH it doesn't really hurt.

**Review by aplaice — APPROVED** [review]
Testing on the AUG repo on Linux (I don't have easy access to Windows):

1. with normal `\n` newlines the patch doesn't break anything.
2. with manually added `\r\n` newlines the patch allows BB to run successfully (without the patch `\r\n` newlines cause BB to crash).

(IMO can/should be merged! :))

**Comment by ohare93** [comment]
Lovely, thanks all. I’ll give it a quick test on Windows tomorrow night, then make a release shortly thereafter.

**Review by ohare93 — APPROVED** [review]

**Pull request merged**